tests/e2e/config_secret_utils.go

```diff
@@ -0,0 +1,331 @@
+	masterIp string, testUser string) error {
+	createRoleCmdFortestUser := loginCmd + "govc role.create CNS-DATASTORE-" + testUser +
+		" Datastore.FileManagement;" +
+		"govc role.create CNS-HOST-CONFIG-STORAGE-" + testUser +
+		" Host.Config.Storage;" +
+		"govc role.create CNS-VM-" + testUser +
+		" VirtualMachine.Config.AddExistingDisk VirtualMachine.Config.AddRemoveDevice;" +
+		"govc role.create CNS-SEARCH-AND-SPBM-" + testUser +
+		" Cns.Searchable StorageProfile.View"
+	framework.Logf(createRoleCmdFortestUser)
+	result, err := sshExec(sshClientConfig, masterIp, createRoleCmdFortestUser)
+	if err != nil && result.Code != 0 {
+		fssh.LogResult(result)
+		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			createRoleCmdFortestUser, masterIp, err)
+	}
+	return nil
+}
+
+func getDataCenterClusterHostAndVmDetails(loginCmd string, sshClientConfig *ssh.ClientConfig,
+	masterIp string) (string, string, error) {
+	// fetch datacenter details
+	ginkgo.By("Executing command for fetching data center details")
+	dataCenter := loginCmd + "govc datacenter.info | grep -i Name | awk '{print $2}' | tr -d '\n'"
+	framework.Logf(dataCenter)
+	dcResult, err := sshExec(sshClientConfig, masterIp, dataCenter)
+	if err != nil && dcResult.Code != 0 {
+		fssh.LogResult(dcResult)
+		return "", "", fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			dataCenter, masterIp, err)
+	}
+
+	// fetch cluster details
+	ginkgo.By("Executing command for fetching cluster details")
+	cluster := loginCmd + "govc ls /" + dcResult.Stdout + "/host | tr -d '\n'"
+	framework.Logf(cluster)
+	clusterResult, err := sshExec(sshClientConfig, masterIp, cluster)
+	if err != nil && clusterResult.Code != 0 {
+		fssh.LogResult(clusterResult)
+		return "", "", fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			cluster, masterIp, err)
+	}
+
+	// Get all esxi hosts details
+	ginkgo.By("Executing command for fetching esxi host details")
+	hosts := loginCmd + "govc ls " + clusterResult.Stdout + " " + " | grep 10. > /root/hosts.txt | tr -d '\n'"
+	framework.Logf(hosts)
+	hostsResult, err := sshExec(sshClientConfig, masterIp, hosts)
+	if err != nil && hostsResult.Code != 0 {
+		fssh.LogResult(hostsResult)
+		return "", "", fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			hosts, masterIp, err)
+	}
+
+	// Get all k8s vm details
+	ginkgo.By("Executing command for fetching k8s vm details")
+	vms := loginCmd + "govc ls /" + dcResult.Stdout + "/vm" + " " + "| grep k8s > /root/k8s_vm.txt | tr -d '\n'"
+	framework.Logf(vms)
+	vMsResult, err := sshExec(sshClientConfig, masterIp, hosts)
+	if err != nil && vMsResult.Code != 0 {
+		fssh.LogResult(vMsResult)
+		return "", "", fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			vms, masterIp, err)
+	}
+	return dcResult.Stdout, clusterResult.Stdout, nil
+}
+
+func setDataCenterLevelPermission(loginCmd string, sshClientConfig *ssh.ClientConfig,
```

Comment:
in all the `set*Permission()` methods, get user, permission and role to set
that way we can reuse the methods

Reply:
done